src/EnergyPlus/GroundHeatExchangers.cc

```diff
@@ -1532,18 +1507,14 @@ void GLHESlinky::calcGFunctions(EnergyPlusData &state)
     // SUBROUTINE INFORMATION:
     //       AUTHOR:          Matt Mitchell
     //       DATE WRITTEN:    February, 2015
-    //       MODIFIED         na
-    //       RE-ENGINEERED    na
 
     // PURPOSE OF THIS SUBROUTINE:
     // calculates g-functions for the slinky ground heat exchanger model
 
     // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
-    Real64 tLg_max(0.0);
     constexpr Real64 tLg_min(-2);
     constexpr Real64 tLg_grid(0.25);
     constexpr Real64 ts(3600);
-    Real64 tLg;
     constexpr Real64 convertYearsToSeconds(356 * 24 * 60 * 60);
     Real64 fraction;
     Array2D<Real64> valStored({0, this->numTrenches}, {0, this->numCoils}, -1.0);
```

Comment:
This array is 0-indexed? Then why does it go to `numTrenches` and not to `numTrenches-1`?

Reply:
for (int m1 = 1; m1 <= numRC; ++m1) {
            for (int n1 = 1; n1 <= numLC; ++n1) {
                for (int m = 1; m <= this->numTrenches; ++m) {
                    for (int n = 1; n <= this->numCoils; ++n) {

                        // Save mm1 and nn1
                        int mm1 = std::abs(m - m1);
                        int nn1 = std::abs(n - n1);

                                valStored(mm1, nn1) = doubleIntegralVal;

                    } // n
                }     // m
            }         // n1
        }             // m1